src/Features/CSharp/Portable/SplitOrMergeIfStatements/CSharpIfLikeStatementGenerator.cs

```diff
@@ -0,0 +1,151 @@
+        {
+            var ifStatement = (IfStatementSyntax)ifLikeStatement;
+
+            var builder = ImmutableArray.CreateBuilder<SyntaxNode>();
+
+            while (ifStatement.Else?.Statement is IfStatementSyntax elseIfStatement)
+            {
+                builder.Add(elseIfStatement);
+                ifStatement = elseIfStatement;
+            }
+
+            if (ifStatement.Else != null)
+            {
+                builder.Add(ifStatement.Else);
+            }
+
+            return builder.ToImmutable();
+        }
+
+        public SyntaxNode WithCondition(SyntaxNode ifLikeStatement, SyntaxNode condition)
+        {
+            var ifStatement = (IfStatementSyntax)ifLikeStatement;
+            return ifStatement.WithCondition((ExpressionSyntax)condition);
+        }
+
+        public SyntaxNode WithStatement(SyntaxNode ifLikeStatement, SyntaxNode statement)
+        {
+            var ifStatement = (IfStatementSyntax)ifLikeStatement;
+            return ifStatement.WithStatement(SyntaxFactory.Block((StatementSyntax)statement));
+        }
+
+        public SyntaxNode WithStatementsOf(SyntaxNode ifLikeStatement, SyntaxNode otherIfLikeStatement)
+        {
+            var ifStatement = (IfStatementSyntax)ifLikeStatement;
+            var otherIfStatement = (IfStatementSyntax)otherIfLikeStatement;
+            return ifStatement.WithStatement(otherIfStatement.Statement);
+        }
+
+        public SyntaxNode WithElseLikeClausesOf(SyntaxNode ifStatement, SyntaxNode otherIfStatement)
+        {
+            return ((IfStatementSyntax)ifStatement).WithElse(((IfStatementSyntax)otherIfStatement).Else);
+        }
+
+        public SyntaxNode ToIfStatement(SyntaxNode ifLikeStatement)
+            => ifLikeStatement;
+
+        public SyntaxNode ToElseIfClause(SyntaxNode ifLikeStatement)
+            => ((IfStatementSyntax)ifLikeStatement).WithElse(null);
+
+        public void InsertElseIfClause(SyntaxEditor editor, SyntaxNode afterIfLikeStatement, SyntaxNode elseIfClause)
+        {
+            var ifStatement = (IfStatementSyntax)afterIfLikeStatement;
+            var elseIfStatement = (IfStatementSyntax)elseIfClause;
+
+            var newElseIfStatement = elseIfStatement.WithElse(ifStatement.Else);
+            var newIfStatement = ifStatement.WithElse(SyntaxFactory.ElseClause(newElseIfStatement));
+
+            if (ifStatement.Else == null && ContainsEmbeddedIfStatement(ifStatement))
+            {
+                // If the if statement contains an embedded if statement (not wrapped inside a block), adding an else
+                // clause might introduce a dangling else problem (the 'else' would bind to the inner if statement),
+                // so if there used to be no else clause, we'll insert a new block to prevent that.
+                newIfStatement = newIfStatement.WithStatement(SyntaxFactory.Block(newIfStatement.Statement));
+            }
+
+            editor.ReplaceNode(ifStatement, newIfStatement);
```

Comment:
curious how the VB version works.  if not necessary, it would be great if this could be pulled into the caller.

Reply:
It cannot. In C#, we have to replace the if statement, and that does the trick (transfers all the following else clauses). In VB, we have to insert a new else-if into a syntax list of else-ifs.